Supplier Input Implementation

tremainebuchanan opened the pull request:
User now able to create an input by clicking the global add button, selecting input, completing the form and submitting it.

tremainebuchanan committed:
Commitment before updating develop branch

tremainebuchanan committed:
Merge branch 'call-report-impl' of https://github.com/slashroots/jas-mis-app into call-report-impl

tremainebuchanan committed:
Call Input dialog implemented.
Designing the layout of the form to remain consistent with
application design.

tremainebuchanan committed:
Farmer Call Input form implemented.
The function would need to be refracted to be more
general purpose to handle gracefully all other call
types.

tremainebuchanan committed:
Reimplemented call creation.
Prior implementation was creating a bug once the user
tried to create a call from a farmer profile page or
a buyer profile page. The dialog box's controller is
aliased from the Search Ctrl hence calling a function
to use the dialog resulted in $scope having
incorrect references to objects not needed in the
current scope.

tremainebuchanan committed:
Minor call log fix completed.

tremainebuchanan committed:
Calls associated with a buyer or farmer now visible on profile page.

tremainebuchanan committed:
Bug found while rendering call report on dashboard fixed.
Added call route to provide an expanded view of all calls
received by user.

tremainebuchanan committed:
Universal call feature implemented.
The user is now able to create a call from the
Floating action button menu. For now, the cc_entity_id
is a randomly generated ObjectID.

tremainebuchanan committed:
Supply input dialog created

tremainebuchanan committed:
Call Creation functionality revamped. Calls created are now apart of the entity as a subdocument.

tremainebuchanan committed:
Call Addition functionality complete. The user is now able to:
1. Create a call for either a farmer, buyer or other.
2. Create a call by clicking the univeral add button,
then selecting Call.
3. Create a call by clicking the call icon on either
a farmer or buyer small panel once the user searches
for the entity by name.
Each call is stored in an array "calls" in either a
farmer or buyer document as a sub document.

tremainebuchanan committed:
Header added to all call dialog boxes

tremainebuchanan committed:
Header added to supplier input dialog

tremainebuchanan committed:
Switching to admin-impl branch

tremainebuchanan committed:
Minor modifications done on branch.

matjames007 committed:
I made adjustments to the sorting of the calls.  This now sorts dates by default.  The newest calls move to the top of the lists throughout the application.  Also a bug has been discovered during the creation of a call from the farmer profile page.  More will be documented in an issue.

-Matthew Budram

matjames007 committed:
Did a bit of date formatting to the call logs. Also, in navigation I have a bit of logic that clears the form in the case of success/cancellation.

matjames007 committed:
I am attempting the close issue #85.  The entity name should now be displayed irrespective of the type of entity being store on the database.

Also the call log for the farmer has been adjusted and beautified.  Dates have been formatted and sorted.

matjames007 committed:
Minor changes to the UI of the call logs.

matjames007 commented:
@tremainebuchanan run through the code and try and clean up the TODOs specific to this pull request.

tremainebuchanan committed:
Completed TODO's in calls.js file

tremainebuchanan committed:
TODOs to restrict/limit queries to backend not complete.
To implement restrictions/limits on GET requests to the server,
this would be done on the backend. An issue will be created in order
to fix same. Calls Listing page modified to inlcude headers for
each column.

tremainebuchanan commented:
@matjames007 can you review this pull request and merge same into develop. Thanks. Another branch that was created from this branch will be pushed between today and tomorrow.
